### ISSUE TITLE
usage with eslint-loader, warnings/errors are swallowed after successful compilation

### ISSUE THREAD

**Issue by kelly-tock** [webpack-contrib/cache-loader#72]
* Operating System: OS X
* Node Version: 11.12.0
* NPM Version: 6.7.0
* webpack Version: 4.30.0
* cache-loader Version: 3.0.0

### Expected Behavior

with eslint-loader set up to not fail on errors/warnings, and if cache loader is used with babel-loader, warnings/errors should show up consistently. 
- on initial startup, should see errors/warnings
- after stopping, then starting again, should see the same errors

### Actual Behavior

with eslint-loader set up to not fail on errors/warnings, and if cache loader is used with babel-loader, warnings/errors should show up consistently. 
- on initial startup, see errors/warnings
- after stopping, then starting again, the same errors/warnings are not shown

### Code

```js
 {
        enforce: 'pre',
        test: /\.(ts|tsx)$/,
        include: [path.join(cwd, 'src/js'), path.join(cwd, '../shared/src/js')],
        loader: 'eslint-loader',
        options: {
          configFile: path.resolve(cwd, './.eslintrc.js'),
          emitWarning: true,
          failOnWarning: false,
        },
      },
      {
        include: [path.join(cwd, 'src/js'), path.join(cwd, '../shared/src/js')],
        test: /\.(ts|tsx)$/,
        use: [
          {
            loader: 'cache-loader'
          },
          {
            loader: 'babel-loader',
            options: {
              cacheDirectory: true,
              cacheCompression: false,
              compact: false,
              babelrc: false,
              plugins: ['@babel/plugin-syntax-dynamic-import'],
              presets: ['babel-preset-tock'],
            },
          },
        ],
      },
```
### How Do We Reproduce?

introduce a warning from linting, such as no-unused-variables, startup webpack with webpack dev server, and see the warning. 
stop webpack dev server, then start again. you will not see the error.

**Comment by barberdt** [comment]
Repro case provided here: https://github.com/webpack-contrib/eslint-loader/issues/267#issuecomment-528438238.

As mentioned in that comment, not sure which package this issue truly belongs to. Happy to provide the repro information wherever needed.

Thanks for any and all support on this!

**Comment by evilebottnawi** [comment]
You don't need cache-loader for eslint-loader, use preLoader for eslint-loader

**Comment by barberdt** [comment]
@evilebottnawi no, but we need `cache-loader` for other reasons. The repro repository provided is as bare-bones as necessary to reproduce the issue. In our private corporate repository, we are using `cache-loader` (among many other dependencies) for a variety of other reasons.

**Comment by rvillalba-novetta** [comment]
I'm seeing similar issues with ts-loader. On first run I have errors reported, on second run it just reports success and grabs what's in the cache.
